airbyte-integrations/connectors/source-google-ads/unit_tests/test_source.py

```diff
@@ -398,8 +398,11 @@ def test_checkConnection_should_pass_when_configValid(mocker):
 
 
 def test_checkConnection_should_fail_when_apiCallFails(mocker):
-    mocker.patch("source_google_ads.google_ads.GoogleAdsClient", return_value=MockErroringGoogleAdsClient)
+    # We patch the object inside source.py because that's the calling context
+    # https://docs.python.org/3/library/unittest.mock.html#where-to-patch
+    mocker.patch("source_google_ads.source.GoogleAds", MockErroringGoogleAdsClient)
```

Comment:
this is what the PR was mostly tripping up on: 

1. the patch needed to happen in the calling context i.e: you needed to patch the `GoogleAds` client object inside `source.py` not `google_ads.py` ( ya know...if it wasn't giga obvious :P ) 
1. using `return_value` was screwing things up. Using this positional argument (it could just as well be called `new=..`) sets the `new` attribute in this function call, which is the right one to use for reasons I don't 100% understand (I think `return_value` is the right way to mock function return values?)

In conclusion, wtf python